app/main.py

```diff
@@ -1,2 +1,57 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.hash_table = self.create_hash_table()
+        self.load_factor = self.calculate_factor()
+        self.size = 0
+
+    def create_hash_table(self) -> list:
+        return [None for _ in range(self.capacity)]
+
+    def calculate_factor(self) -> int:
+        return int(self.capacity * (2 / 3))
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        key_hash = hash(key)
+        position = key_hash % self.capacity
+        while True:
+            if not self.hash_table[position]:
+                self.hash_table[position] = (key, value, key_hash)
+                self.size += 1
+                if self.size >= self.load_factor:
```

Comment:
Also look , your logic is repeating in the different methods ( same while loop , almost same cheks etc ) . Try to create some func to reuse your logic